### PR TITLE
implement readlengths

### PR DESCRIPTION
**Pull request by jeffkaufman**
Implement `readlengths` pipeline stage.  See `README` for the goal of the stage and how it works (and if the readme bit isn't clear enough ask me to make it clearer!)

### CODE REVIEW

**Comment by jeffkaufman** [comment]
@lennijusten can you take a look when you get a chance?

File: run.py [hunk 1]
```diff
@@ -638,7 +638,87 @@ def taxid_matches(taxid, category):
             "%s/%s/samplereads/" % (S3_BUCKET, args.bioproject)])
 
 def readlengths(args):
-   pass
+   available_samplereads_inputs = get_files(args, "samplereads")
```
Comment (lennijusten) [inline run.py:641]: Why perform read-length calculations on a sub-sample of reads if you are still looping through the entire `.collapsed.` file in `available_cleaned_inputs`?

I would think `if title not in target_read_ids: continue` isn't that much faster than getting the sequence length. Is the idea to have a consistent sub-sample to do other calculations on as well?

**Review by lennijusten — APPROVED** [review]
Looks good! Left one clarifying question. Sorry I didn't see the review request earlier.

Also reminder to remove the `Not implemented yet!` in the README for readlengths.

File: run.py (hunk 1 above)
Comment (jeffkaufman) [reply run.py:641]: The reason for subsampling is that a list of every read id matching a broad criteria like "is bacterial" would be large enough that I wouldn't want to store it in memory while scanning the fastq files for matches.

For example, if we have 100M read pairs in a file, and ids can be pretty long (say 100 bytes) so a list of all the read ids would be >10GB (and probably more than that: python strings have ~50 bytes of overhead, then there's overhead for storing it in a set)

**Comment by jeffkaufman** [comment]
> Also reminder to remove the Not implemented yet! in the README for readlengths.

Thanks; done!

**Pull request merged**